Apps/GatewayApi/test/unit/Services.Test/DelegationServiceTests.cs

```diff
@@ -80,6 +95,91 @@ public static IEnumerable<object[]> TestCases()
             yield return new object[] { ValidEmail, ValidNickname, 1, InvalidDataSources, false, InvalidDataSourceErrorMessage };
         }
 
+        /// <summary>
+        /// Should associate a delegation.
+        /// </summary>
+        /// <param name="resourceOwnerHdid">The delegation's resource owner hdid.</param>
+        /// <param name="delegateHdid">The delegate's hdid.</param>
+        /// <param name="profileHdid">The profile hdid stored in Delegation.</param>
+        /// <param name="secondsToExpiration">The number of seconds to expiration.</param>
+        /// <param name="success">The value indicates whether the test should succeed or not.</param>
+        /// <param name="errorMessage">The error message to display when success is false.</param>
+        [Theory]
+        [InlineData(ResourceOwnerHdid, DelegateHdid, null, 0, true, null)] // 0 Hours
+        [InlineData(ResourceOwnerHdid, DelegateHdid, null, -172800, false, SharingLinkExpirationErrorMesage)] // 48 hours
+        [InlineData(ResourceOwnerHdid, DelegateHdid, null, -172801, false, SharingLinkExpirationErrorMesage)] // 48 hours plus 1 second
+        [InlineData(ResourceOwnerHdid, ResourceOwnerHdid, null, 0, false, DelegationSelfAssociationErrorMessage)]
+        [InlineData(ResourceOwnerHdid, DelegateHdid, ProfileHdid, 0, false, DelegationAlreadyAssociatedErrorMessage)]
+        public async Task AssociateDelegation(string resourceOwnerHdid, string delegateHdid, string? profileHdid, int secondsToExpiration, bool success, string? errorMessage)
+        {
+            // Arrange
+            TimeZoneInfo localTimezone = DateFormatter.GetLocalTimeZone(GetConfiguration());
+            Mock<IDelegationDelegate> delegationDelegate = new();
+            PatientDetails patient = GetPatient();
+
+            int expiryHours = int.Parse(ExpiryHours);
+            DateTime createdDateTime = DateTime.UtcNow.AddSeconds(secondsToExpiration);
+
+            Delegation delegation = new()
+            {
+                Id = Guid.NewGuid(),
+                ResourceOwnerHdid = resourceOwnerHdid,
+                ProfileHdid = profileHdid,
+                CreatedDateTime = createdDateTime,
+
+                ResourceOwnerIdentifier = $"{patient.PreferredName.GivenName} {patient.PreferredName.Surname[0]}",
+                Nickname = ValidNickname,
+                ExpiryDate = DateOnly.FromDateTime(TimeZoneInfo.ConvertTimeFromUtc(DateTime.UtcNow, localTimezone)).AddDays(30),
```

Comment:
We can use UTC for the ExpiryDate in Delegation, to more closely match how they would come from the DB.